### ISSUE TITLE
Add read-only role

### ISSUE THREAD

**Issue by aharwood2** [UoMResearchIT/CapX#441]
David Benko would like read-only access to:

- [x] Projects
- [x] Project Details (not notes!)
- [x] Capacity

Can we create a read-only role that permits this?

**Comment by aharwood2** [comment]
It has occurred to me that we've missed something here. The way the `Role` and `Person` entities work together is it assumes that every role is attached to a person with every person being a member of the team and hence an RSE who can be assignable. For readers this won't be the case. We are going to have to look at either:

a) allowing a role to not be attached to a person and guard against it in the code every time we assume that it must (probably the easiest option)
b) do something with the person entity so you can have different types of people in the DB and modify the interfaces and the person service etc. to cater for it (probably hard).

**Comment by aharwood2** [comment]
Bumping to the next milestone as it requires some further work.

**Comment by aharwood2** [comment]
As per our conversation last we, we just need to check that any time we access the `Person` object via the `Role` object, we don't automatically assume that it has to be not null. It may be that it occurs on pages that the Reader role can't access anyway but if not then we need to check before we can regard this as complete. I think the `ProjectDetails` pages crashed when I was logged in as a user with Reader permissions so that is one area we might need to look at?

**Comment by aharwood2** [comment]
@fdq09eca -- is this now ready for another PR?
